trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/expressiontranslation/BitabsTranslation.java

```diff
@@ -418,4 +394,56 @@ private Expression constructShiftWithLiteralOptimization(final ILocation loc, fi
 		final Expression shiftFactorExpr = mTypeSizes.constructLiteralForIntegerType(loc, typeRight, shiftFactorBigInt);
 		return ExpressionFactory.newBinaryExpression(loc, op1, left, shiftFactorExpr);
 	}
+
+	private Expression applyWraparoundIfNecessary(final ILocation loc, final Expression expr, final CPrimitive type) {
+		if (!mTypeSizes.isUnsigned(type)) {
+			return expr;
+		}
+		final BigInteger maxValuePlusOne = mTypeSizes.getMaxValueOfPrimitiveType(type).add(BigInteger.ONE);
+		return ExpressionFactory.newBinaryExpression(loc, BinaryExpression.Operator.ARITHMOD, expr,
+				ExpressionFactory.createIntegerLiteral(loc, maxValuePlusOne.toString()));
+	}
+
+	private static boolean isZero(final Expression expr) {
+		return expr instanceof IntegerLiteral && "0".equals(((IntegerLiteral) expr).getValue());
```

Comment:
Like above, here's another place to consider `TypeSizes::extractIntegerValue`

Reply:
See my other comment. It should not really make a difference.